testing/test_subtests.py

```diff
@@ -0,0 +1,840 @@
+        expected_lines += ["* 1 passed, 2 subtests passed, 3 subtests xfailed in *"]
+        result.stdout.fnmatch_lines(expected_lines)
+
+    def test_typing_exported(
+        self, pytester: pytest.Pytester, mode: Literal["normal", "xdist"]
+    ) -> None:
+        pytester.makepyfile(
+            """
+            from pytest import SubTests
+
+            def test_typing_exported(subtests: SubTests) -> None:
+                assert isinstance(subtests, SubTests)
+            """
+        )
+        if mode == "normal":
+            result = pytester.runpytest()
+            expected_lines = ["collected 1 item"]
+        else:
+            assert mode == "xdist"
+            pytest.importorskip("xdist")
+            result = pytester.runpytest("-n1", "-pxdist.plugin")
+            expected_lines = ["1 worker [1 item]"]
+        expected_lines += ["* 1 passed *"]
+        result.stdout.fnmatch_lines(expected_lines)
+
+    def test_no_subtests_reports(
+        self, pytester: pytest.Pytester, mode: Literal["normal", "xdist"]
+    ) -> None:
+        pytester.makepyfile(
+            """
+            import pytest
+
+            def test_foo(subtests):
+                for i in range(5):
+                    with subtests.test(msg="custom", i=i):
+                        pass
+        """
+        )
+        # Without `--no-subtests-reports`, subtests are reported normally.
+        result = pytester.runpytest("-v")
+        result.stdout.fnmatch_lines(
+            [
+                "*collected 1 item*",
+                "test_no_subtests_reports.py::test_foo * (i=0) SUBPASS*",
+                "*test_no_subtests_reports.py::test_foo PASSED*",
+                "* 1 passed, 5 subtests passed in*",
+            ]
+        )
+
+        # With `--no-subtests-reports`, passing subtests are no longer reported.
+        result = pytester.runpytest("-v", "--no-subtests-reports")
+        result.stdout.fnmatch_lines(
+            [
+                "*collected 1 item*",
+                "*test_no_subtests_reports.py::test_foo PASSED*",
+                "* 1 passed in*",
+            ]
+        )
+        result.stdout.no_fnmatch_line("*SUBPASS*")
+
+        # Rewrite the test file so the tests fail. Even with the flag, failed subtests are still reported.
+        pytester.makepyfile(
+            """
+            import pytest
+
+            def test_foo(subtests):
+                for i in range(5):
+                    with subtests.test(msg="custom", i=i):
+                        assert False
+        """
+        )
+        result = pytester.runpytest("-v", "--no-subtests-reports")
+        result.stdout.fnmatch_lines(
+            [
+                "*collected 1 item*",
+                "test_no_subtests_reports.py::test_foo * (i=0) SUBFAIL*",
+                "*test_no_subtests_reports.py::test_foo PASSED*",
+                "* 5 failed, 1 passed in*",
+            ]
+        )
+
+
+class TestSubTest:
+    """Test.subTest functionality."""
```

Comment:
```suggestion
    """Test subTest() functionality in unittest tests."""
```

Unless I misunderstood.